contributors/design-proposals/daemonset-update.md

```diff
@@ -0,0 +1,222 @@
+# DaemonSet Updates
+
+**Author**: @madhusudancs, @lukaszo, @janetkuo
+
+**Status**: Proposal
+
+## Abstract
+
+A proposal for adding the update feature to `DaemonSet`. This feature will be
+implemented on server side (in `DaemonSet` API). 
+
+Users already can update a `DaemonSet` today (Kubernetes release 1.5), which will
+not cause changes to its subsequent pods, until those pods are killed. In this
+proposal, we plan to add a "RollingUpdate" strategy which allows DaemonSet to
+downstream its changes to pods. 
+
+## Requirements
+
+In this proposal, we design DaemonSet updates based on the following requirements:
+
+- Users can have a controlled number of in-flight updates of a DaemonSet (e.g.
+  DaemonSet only updates one pod at a time)
+- Users can monitor the status of a DaemonSet update (e.g. the number of pods
+  that are updated and healthy)
+- A broken DaemonSet update should not continue 
+- Users should be able to update a DaemonSet, even during an ongoing DaemonSet
+  upgrade (e.g. update the DaemonSet to fix a broken DaemonSet update)
+- Users should be able to view the history of previous DaemonSet updates 
+- Users can figure out the revision of a DaemonSet's pod (e.g. which version is
+  this DaemonSet pod?)
+
+Here are some potential requirements that haven't been covered by this proposal:
+
+- Users can have rate-limited DaemonSet updates (e.g. only upgrade 10% per hour)
+- DaemonSet should provide at-most-one guarantee per node (i.e. at most one pod
+  from a DaemonSet can exist on a node at any time)
```

Comment:
DS controller should have at-most-one by design. However, because it's listing pods using a pod cache store it's possible that more than one daemon pod is created on a node temporarily. 

If we choose to start with at-least-one that means we create a new daemon pod before killing the old one during an update?